### PR TITLE
fix(mock-ase): interact bug by updating idp secret

### PR DESCRIPTION
**Pull request by BlairCurrey**
## Changes proposed in this pull request

- updates `x-idp-secret` header in `mock-account-servicing-entity` to fix interact bug

## Context

fixes https://github.com/interledger/rafiki/issues/2745

## Checklist

- [ ] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass
- [ ] Bruno collection updated

### CODE REVIEW

File: localenv/mock-account-servicing-entity/app/lib/apiClient.ts [hunk 1]
```diff
@@ -8,6 +8,8 @@ export const StepNames = {
   endInteraction: 3
 }
 
+const IDP_SECRET = '2pEcn2kkCclbOHQiGNEwhJ0rucATZhrA807HTm2rNXE='
```
Comment (mkurapov) [inline localenv/mock-account-servicing-entity/app/lib/apiClient.ts:11]: can we pass this through `process.env`?

Comment (BlairCurrey) [reply localenv/mock-account-servicing-entity/app/lib/apiClient.ts]: it was a bit involved but yes [15178f6](https://github.com/interledger/rafiki/pull/2746/commits/15178f6f23941359e450d97199c3aa4d5845696a)

**Comment by golobitch** [comment]
Yes, apparently I introduced this bug when working on auth env variables. Thank you for spotting and fixing this @BlairCurrey

File: localenv/mock-account-servicing-entity/app/lib/parse_config.server.ts [hunk 2]
```diff
@@ -3,6 +3,10 @@ import { readFileSync } from 'fs'
 import { loadOrGenerateKey } from '@interledger/http-signature-utils'
 import type { Config } from 'mock-account-service-lib'
 
+if (!process.env.IDP_SECRET) {
+  throw new Error('environment variable IDP_SECRET is required')
+}
```
Comment (mkurapov) [inline localenv/mock-account-servicing-entity/app/lib/parse_config.server.ts:8]: Should we do the same for the other env vars? Most of them seem mandatory

File: localenv/mock-account-servicing-entity/app/routes/mock-idp._index.tsx [hunk 3]
```diff
@@ -1,5 +1,13 @@
 import ConsentScreen from '~/routes/consent-screen'
+import { json } from '@remix-run/node'
+import { useLoaderData } from '@remix-run/react'
+import { CONFIG } from '~/lib/parse_config.server'
+
+export function loader() {
+  return json({ idpSecret: CONFIG.idpSecret })
+}
```
Comment (mkurapov) [inline localenv/mock-account-servicing-entity/app/routes/mock-idp._index.tsx:8]: Just thinking out loud, should some of these requests (ie, those requested via ApiClient.*) be handled in the `action()` handler on the server side instead? Maybe I'm just not used to remix, but seems a bit odd to have to return a env variable (that's probably best to keep secret) to the front-end to be used for admin API requests.

**Review by mkurapov — APPROVED** [review]
Non blocking comments, just worth consideration

**Pull request merged**